cibuildwheel/__main__.py

```diff
@@ -187,17 +190,23 @@ def main() -> None:
     if not output_dir.exists():
         output_dir.mkdir(parents=True)
 
-    with cibuildwheel.util.print_new_wheels(
-        "\n{n} wheels produced in {m:.0f} minutes:", output_dir
-    ):
-        if platform == "linux":
-            cibuildwheel.linux.build(options)
-        elif platform == "windows":
-            cibuildwheel.windows.build(options)
-        elif platform == "macos":
-            cibuildwheel.macos.build(options)
-        else:
-            assert_never(platform)
+    tmp_path_string = mkdtemp(prefix="cibw-run-")
+    try:
+        tmp_path = Path(tmp_path_string).resolve(strict=True)
+        with cibuildwheel.util.print_new_wheels(
+            "\n{n} wheels produced in {m:.0f} minutes:", output_dir
+        ):
+            if platform == "linux":
+                cibuildwheel.linux.build(options, tmp_path)
+            elif platform == "windows":
+                cibuildwheel.windows.build(options, tmp_path)
+            elif platform == "macos":
+                cibuildwheel.macos.build(options, tmp_path)
+            else:
+                assert_never(platform)
+    finally:
+        # TODO, log path that were not removed
```

Comment:
remains to log removal failures
probably only worth it on local builds ?

Reply:
No strong opinion here, but always logging removal failures makes sense to me. It _shouldn't_ fail, it probably fails due to something external to cibuildwheel. So we should probably log it in case it ever gets worse in the future.

Reply:
It now emits a warning but doesn't log all remaining files (could grow out of proportion or would need some non trivial tricks to limit the log length)